monai/apps/auto3d/data_analyzer.py

```diff
@@ -0,0 +1,644 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Step 1 of the AutoML pipeline. The dataset is analysized with this script.
+"""
+
+import argparse
+import copy
+import logging
+import time
+import warnings
+from functools import partial
+from typing import Dict, List, Union
+
+import numpy as np
+import torch
+
+from monai import data, transforms
+from monai.apps.auto3d.data_utils import datafold_read, recursive_getkey, recursive_getvalue, recursive_setvalue
+from monai.data.meta_tensor import MetaTensor
+from monai.utils import min_version, optional_import
+
+yaml, _ = optional_import("yaml")
+tqdm, has_tqdm = optional_import("tqdm", "4.47.0", min_version, "tqdm")
+measure, _ = optional_import("scipy.ndimage.measurements")
+
+logger = logging.getLogger("global")
+logger.addHandler(logging.StreamHandler())
+
+__all__ = ["DataAnalyzer"]
+
+
+class DataAnalyzer:
+    """
+    The DataAnalyzer automatically analyzes a given medical image datasets and reports out the statistics.
```

Comment:
I think it's a useful generic implementation, shall we consider https://github.com/Project-MONAI/MONAI/issues/4824 here, and perhaps move this to `monai.data` and deprecate `monai.data.dataset_summary`?

Reply:
Good point, @yiheng-wang-nv @dongyang0122 What do you think?

Thanks.

Reply:
Can we make another PR if we decide to move it under monai.data and deprecate monai.data.dataset_summary?